### DISCUSSION TITLE
Data preloading with getServerSideProps

### DISCUSSION THREAD

**Discussion by saibem** [vercel/next.js#47204] [Help]
### Summary

Hello, I'm new to Next.js and I require assistance with preloading data before its rendering. I am utilizing getServerSideProps to invoke the fetch function and subsequently dispatch the received data to the Redux store. However, the issue I am facing is that getServerSideProps is not being called. Thanks :)

### Additional information

```js
Here is my code:

export const getServerSideProps: GetServerSideProps = async () => {
  console.log('getServerSideProps')
  const { dispatch } = store;
  await preloadFn(dispatch);

  return {
    props: {},
  };
};


export const preloadFn = async (
    dispatch: ThunkDispatch<StoreStateAll, undefined, AllActions>
) => {
    const navbarData: NavbarApiResponse = await fetch(
        "https://example.com/2"
    ).then((response) => response.json());

    dispatch(fetchNavbarSuccess(navbarData));
    const footerData: FooterData = await fetch(
        "https://example.com/1"
    ).then((response) => response.json());
    // Dispatch footer data to Redux store
    dispatch(fetchFooterSuccess(footerData));
};
```

**Comment by icyJoseph** [comment]
GetServerSideProps (GSSP) is a special kind of function.

During first time navigation, that is, arriving to a page, GSSP runs on the server, and its result is passed as props to the page component, then the page is rendered on the server, including all HTML tags, scripts, and links, and the entire HTML document is sent to the client. You can call this a dehydrated page.

Once on the client, something called hydration takes place, which using the data built into the HTML document, walks the DOM adding interactivity, and essentially recreates the UI. 

When you do an SPA transition into a page with GSSP, that is from within your application, Next.js sends a request to the server, where GSSP is executed, and its results are packed as JSON, and sent back to the client. Once in the client this JSON is parsed and used as props to the page component.

> In Vercel deployments GSSP is run as a serverless function, a lambda, which makes it super cheap to manage!

So, dispatching to the Redux store, from within GSSP, feels incorrect. 

- For first navigation situations, you might be able to dispatch to the store, but you should just pass that data onto the store initializer, within _app, or the page itself like that
   - So the initialization (preload) logic won't run during hydration, and it'll most likely crash out on divergent HTML, because the store won't be up-to-date
- During SPA navigations, GSSP runs entirely server side, so there's no store what so ever, that can listen to the dispatched actions.

Perhaps in the first case, you could hack it around, though I'd expect it to fails anyway. The second case is definitely breaking.

This person has some code you could try to use: https://github.com/vercel/next.js/discussions/20122#discussioncomment-199302
